source/Lib/Utilities/ThreadPool.cpp

```diff
@@ -210,14 +212,18 @@ void ThreadPool::threadProc( int threadId )
           }
 
           if( !l.owns_lock()
-              && m_waitingThreads.load( std::memory_order_relaxed ) > 1
               && ( BUSY_WAIT_TIME.count() == 0 || std::chrono::steady_clock::now() - startWait > BUSY_WAIT_TIME )
               && !m_exitThreads )
           {
             ITT_TASKSTART( itt_domain_thrd, itt_handle_TPblocked );
+            ScopeIncDecCounter cntr( m_poolPause.m_waitingForLockThreads );
             l.lock();
             ITT_TASKEND( itt_domain_thrd, itt_handle_TPblocked );
           }
+          else if (std::chrono::steady_clock::now() - startWait > std::chrono::milliseconds(500))
```

Comment:
The 500 ms is just a random number I put here. This could also be bigger or smaller. I just found that pausing the pool (waiting for the conditional variable) is a bit slow and costs a bit of performance if done too often. So I wanted to make this a last resort when waiting too long.

Reply:
That's fine and 500ms sounds like a sensible duration.